flow360/component/geometry.py

```diff
@@ -208,7 +218,97 @@ def from_file(
         # For type hint only but proper fix is to fully abstract the Draft class too.
         return super().from_file(file_names, name, solver_version, length_unit, tags)
 
-    def _get_metadata(self):
-        # get the metadata when initializing the object (blocking)
-        # My next PR
-        pass
+    def _get_metadata(self) -> None:
+        "Blockingly trying to download the metadata and populate the metadata info"
+        start_time = time.time()
+        while self._webapi.status == Flow360Status.PROCESSED:
+            if time.time() - start_time > TIMEOUT_MINUTES * 60:
+                raise TimeoutError(
+                    "Timeout: Process did not finish within the specified timeout period"
+                )
+            time.sleep(10)
+        print("self._webapi.status = ", self._webapi.status)
+        log.debug("Metadata pipeline completed, downloading metadata now...")
+        temp_file_path = "_temp_metadata_" + self.id + ".json"
+        try:
+            if os.path.exists(temp_file_path) is False:
+                # pylint: disable=protected-access
+                self._webapi._download_file(
+                    "results/geometry.json",
+                    to_file=temp_file_path,
+                    to_folder=".",
+                    overwrite=True,
+                    progress_callback=None,
+                    stealth_mode=True,
+                )
+            with open(temp_file_path, "r", encoding="utf-8") as f:
+                _meta = json.load(f)
+                self._metadata_info = _GeometryMetadataModel.parse_obj(_meta)
+                assert isinstance(self._metadata_info, pd.BaseModel)
+        finally:
+            os.remove(temp_file_path)
```

Comment:
The issue is that I cannot add unit test without knowing the file name in advance. What I do is I copy a mock metadata.json to the test folder and the `_download_file` will be skipped because file already exists.

Reply:
Is there a way to mock _download_file?

Reply:
yes, see results unit tests for example. They use mock s3 download
https://github.com/flexcompute/Flow360/blob/6ef5253b3ba119fb2f3e6ebefbd1ba9fbf1a4b8e/tests/utils.py#L213

Reply:
Done. Thanks for pointing out,